### PR TITLE
grpc-json: preserve request method in x-envoy-original-method header

### PR DESCRIPTION
**Pull request by JavaScriptBach**
Commit Message: grpc-json: preserve http request method in `x-envoy-original-method` header so that applications have access to it.
Additional Description: The grpc-json transcoder currently forwards HTTP path to applications via "x-envoy-original-path" header. We would find it useful if it also forwarded the HTTP method.
Risk Level: Low
Testing: Updated grpc-json-transcoder unit tests
Docs Changes: Added docs
Release Notes: Added release notes

### CODE REVIEW

File: include/envoy/http/header_map.h [hunk 1]
```diff
@@ -293,6 +293,7 @@ class HeaderEntry {
   HEADER_FUNC(EnvoyRetryGrpcOn)                                                                    \
   HEADER_FUNC(EnvoyRetriableStatusCodes)                                                           \
   HEADER_FUNC(EnvoyRetriableHeaderNames)                                                           \
+  HEADER_FUNC(EnvoyOriginalMethod)                                                                 \
```
Comment (mattklein123) [inline include/envoy/http/header_map.h:296]: Drive by: please don't add this to the O(1) header map, as it only needs to be set.

Also, this change will need documentation and release notes. Thank you!

Comment (JavaScriptBach) [reply include/envoy/http/header_map.h]: Thanks for pointing this out. Could you point out where you would like to see the documentation? I could not find any prior mention of headers in https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/grpc_json_transcoder_filter. Should I create a section there anyway?

Comment (mattklein123) [reply include/envoy/http/header_map.h]: Yeah I think documenting this within the filter config docs would be fine for now. Thank you!

File: docs/root/configuration/http/http_filters/grpc_json_transcoder_filter.rst [hunk 2]
```diff
@@ -87,6 +87,15 @@ can be send by the gRPC server in the server streaming case.
 In this case, HTTP response header `Content-Type` will use the `content-type` from the first
 `google.api.HttpBody <https://github.com/googleapis/googleapis/blob/master/google/api/httpbody.proto>`.
 
+Metadata
```
Comment (lizan) [inline docs/root/configuration/http/http_filters/grpc_json_transcoder_filter.rst:90]: let's call this header. `Metadata` is confusing with [dynamic metadata](https://www.envoyproxy.io/docs/envoy/latest/configuration/advanced/well_known_dynamic_metadata)

Comment (JavaScriptBach) [reply docs/root/configuration/http/http_filters/grpc_json_transcoder_filter.rst]: Done. Can you review again?

**Comment by lizan** [comment]
Can you merge master and resolve conflicts? Then we can merge this. Thank you!

**Comment by JavaScriptBach** [comment]
> Can you merge master and resolve conflicts? Then we can merge this. Thank you!

Done. Thank you!

**Pull request merged**